### PR TITLE
Refacto modals  + improve test

### PR DESCRIPTION
**Pull request by Fuzznimp**
## Description

This PR aims at
- Removing the previous `ConfirmationBox` mess.
- Migrate the modals to tailwindccs.
- Made the test more strict.

I made the use of the `xl` breakpoint to improve the modals on tablet.

I tried to fix the previously named `ClickAwayListener` test but couldn't make it work :/.

## Type of change

- [x] **Chore** (non-breaking change which refactors / improves the existing code base).
- [ ] **Bug fix** (non-breaking change which fixes an issue).
- [ ] **New feature** (non-breaking change which adds functionality).
- [ ] **Breaking change** (fix or feature that would cause existing functionality to not work as expected).

## Checklist

- [x] My code follows the style [**contributing guidelines**][contributing_file] of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.

[contributing_file]: https://github.com/fewlinesco/connect-account/blob/master/README.adoc

### CODE REVIEW

**Comment by frieda-ewlines** [comment]
Task linked: [CU-1ck88hf [tailwind] - Migrate + refacto confirmation boxes](https://app.clickup.com/t/1ck88hf)

**Comment by frieda-ewlines** [comment]
<details><summary>Click here to see the Lighthouse manifest</summary><br><p><strong>/:</strong></p><ul><li>Performance: 0.89</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/:</strong></p><ul><li>Performance: 0.91</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/profile/:</strong></p><ul><li>Performance: 0.94</li><li>Accessibility: 1</li><li>Best practice: 0.93</li><li>SEO: 1</li></ul><hr><p><strong>/account/logins/:</strong></p><ul><li>Performance: 0.92</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/logins/email/new/:</strong></p><ul><li>Performance: 0.91</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/logins/phone/new/:</strong></p><ul><li>Performance: 0.87</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/security/:</strong></p><ul><li>Performance: 0.93</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/security/sudo/:</strong></p><ul><li>Performance: 0.89</li><li>Accessibility: 0.98</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/locale/:</strong></p><ul><li>Performance: 0.87</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr></details>

**Comment by frieda-ewlines** [comment]
<details><summary>Click here to see the Lighthouse manifest</summary><br><p><strong>/:</strong></p><ul><li>Performance: 0.9</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/:</strong></p><ul><li>Performance: 0.94</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/profile/:</strong></p><ul><li>Performance: 0.94</li><li>Accessibility: 1</li><li>Best practice: 0.93</li><li>SEO: 1</li></ul><hr><p><strong>/account/logins/:</strong></p><ul><li>Performance: 0.95</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/logins/email/new/:</strong></p><ul><li>Performance: 0.92</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/logins/phone/new/:</strong></p><ul><li>Performance: 0.93</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/security/:</strong></p><ul><li>Performance: 0.92</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/security/sudo/:</strong></p><ul><li>Performance: 0.92</li><li>Accessibility: 0.98</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/locale/:</strong></p><ul><li>Performance: 0.93</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr></details>

**Comment by frieda-ewlines** [comment]
<details><summary>Click here to see the Lighthouse manifest</summary><br><p><strong>/:</strong></p><ul><li>Performance: 0.86</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/:</strong></p><ul><li>Performance: 0.91</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/profile/:</strong></p><ul><li>Performance: 0.94</li><li>Accessibility: 1</li><li>Best practice: 0.93</li><li>SEO: 1</li></ul><hr><p><strong>/account/logins/:</strong></p><ul><li>Performance: 0.94</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/logins/email/new/:</strong></p><ul><li>Performance: 0.91</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/logins/phone/new/:</strong></p><ul><li>Performance: 0.92</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/security/:</strong></p><ul><li>Performance: 0.92</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/security/sudo/:</strong></p><ul><li>Performance: 0.9</li><li>Accessibility: 0.98</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/locale/:</strong></p><ul><li>Performance: 0.92</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr></details>

**Comment by frieda-ewlines** [comment]
<details><summary>Click here to see the Lighthouse manifest</summary><br><p><strong>/:</strong></p><ul><li>Performance: 0.92</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/:</strong></p><ul><li>Performance: 0.96</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/profile/:</strong></p><ul><li>Performance: 0.95</li><li>Accessibility: 1</li><li>Best practice: 0.93</li><li>SEO: 1</li></ul><hr><p><strong>/account/logins/:</strong></p><ul><li>Performance: 0.96</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/logins/email/new/:</strong></p><ul><li>Performance: 0.94</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/logins/phone/new/:</strong></p><ul><li>Performance: 0.95</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/security/:</strong></p><ul><li>Performance: 0.93</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/security/sudo/:</strong></p><ul><li>Performance: 0.93</li><li>Accessibility: 0.98</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/locale/:</strong></p><ul><li>Performance: 0.94</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr></details>

**Comment by frieda-ewlines** [comment]
<details><summary>Click here to see the Lighthouse manifest</summary><br><p><strong>/:</strong></p><ul><li>Performance: 0.92</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/:</strong></p><ul><li>Performance: 0.96</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/profile/:</strong></p><ul><li>Performance: 0.95</li><li>Accessibility: 1</li><li>Best practice: 0.93</li><li>SEO: 1</li></ul><hr><p><strong>/account/logins/:</strong></p><ul><li>Performance: 0.95</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/logins/email/new/:</strong></p><ul><li>Performance: 0.89</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/logins/phone/new/:</strong></p><ul><li>Performance: 0.94</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/security/:</strong></p><ul><li>Performance: 0.88</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/security/sudo/:</strong></p><ul><li>Performance: 0.92</li><li>Accessibility: 0.98</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/locale/:</strong></p><ul><li>Performance: 0.89</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr></details>

**Comment by frieda-ewlines** [comment]
<details><summary>Click here to see the Lighthouse manifest</summary><br><p><strong>/:</strong></p><ul><li>Performance: 0.93</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/:</strong></p><ul><li>Performance: 0.96</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/profile/:</strong></p><ul><li>Performance: 0.93</li><li>Accessibility: 1</li><li>Best practice: 0.93</li><li>SEO: 1</li></ul><hr><p><strong>/account/logins/:</strong></p><ul><li>Performance: 0.92</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/logins/email/new/:</strong></p><ul><li>Performance: 0.88</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/logins/phone/new/:</strong></p><ul><li>Performance: 0.94</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/security/:</strong></p><ul><li>Performance: 0.91</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/security/sudo/:</strong></p><ul><li>Performance: 0.9</li><li>Accessibility: 0.98</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/locale/:</strong></p><ul><li>Performance: 0.93</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr></details>

**Comment by frieda-ewlines** [comment]
<details><summary>Click here to see the Lighthouse manifest</summary><br><p><strong>/:</strong></p><ul><li>Performance: 0.91</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/:</strong></p><ul><li>Performance: 0.96</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/profile/:</strong></p><ul><li>Performance: 0.95</li><li>Accessibility: 1</li><li>Best practice: 0.93</li><li>SEO: 1</li></ul><hr><p><strong>/account/logins/:</strong></p><ul><li>Performance: 0.96</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/logins/email/new/:</strong></p><ul><li>Performance: 0.93</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/logins/phone/new/:</strong></p><ul><li>Performance: 0.95</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/security/:</strong></p><ul><li>Performance: 0.95</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/security/sudo/:</strong></p><ul><li>Performance: 0.9</li><li>Accessibility: 0.98</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/locale/:</strong></p><ul><li>Performance: 0.92</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr></details>

**Comment by frieda-ewlines** [comment]
<details><summary>Click here to see the Lighthouse manifest</summary><br><p><strong>/:</strong></p><ul><li>Performance: 0.91</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/:</strong></p><ul><li>Performance: 0.96</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/profile/:</strong></p><ul><li>Performance: 0.92</li><li>Accessibility: 1</li><li>Best practice: 0.93</li><li>SEO: 1</li></ul><hr><p><strong>/account/logins/:</strong></p><ul><li>Performance: 0.93</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/logins/email/new/:</strong></p><ul><li>Performance: 0.93</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/logins/phone/new/:</strong></p><ul><li>Performance: 0.94</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/security/:</strong></p><ul><li>Performance: 0.92</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/security/sudo/:</strong></p><ul><li>Performance: 0.9</li><li>Accessibility: 0.98</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/locale/:</strong></p><ul><li>Performance: 0.93</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr></details>

**Comment by frieda-ewlines** [comment]
<details><summary>Click here to see the Lighthouse manifest</summary><br><p><strong>/:</strong></p><ul><li>Performance: 0.94</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/:</strong></p><ul><li>Performance: 0.96</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/profile/:</strong></p><ul><li>Performance: 0.96</li><li>Accessibility: 1</li><li>Best practice: 0.93</li><li>SEO: 1</li></ul><hr><p><strong>/account/logins/:</strong></p><ul><li>Performance: 0.96</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/logins/email/new/:</strong></p><ul><li>Performance: 0.94</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/logins/phone/new/:</strong></p><ul><li>Performance: 0.95</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/security/:</strong></p><ul><li>Performance: 0.93</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/security/sudo/:</strong></p><ul><li>Performance: 0.91</li><li>Accessibility: 0.98</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/locale/:</strong></p><ul><li>Performance: 0.94</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr></details>

File: src/components/modal-overlay.tsx [hunk 1]
```diff
@@ -11,7 +11,8 @@ const ModalOverlay: React.FC<{ onClick?: () => void; className?: string }> = ({
         className
       }
       onClick={onClick}
-    ></div>
+      data-testid="modal-overlay"
```
Comment (Meyclem) [inline src/components/modal-overlay.tsx:14]: Why that name?

**Review by Oluane — APPROVED** [review]

**Comment by frieda-ewlines** [comment]
<details><summary>Click here to see the Lighthouse manifest</summary><br><p><strong>/:</strong></p><ul><li>Performance: 0.84</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/:</strong></p><ul><li>Performance: 0.95</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/profile/:</strong></p><ul><li>Performance: 0.94</li><li>Accessibility: 1</li><li>Best practice: 0.93</li><li>SEO: 1</li></ul><hr><p><strong>/account/logins/:</strong></p><ul><li>Performance: 0.95</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/logins/email/new/:</strong></p><ul><li>Performance: 0.91</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/logins/phone/new/:</strong></p><ul><li>Performance: 0.95</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/security/:</strong></p><ul><li>Performance: 0.92</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/security/sudo/:</strong></p><ul><li>Performance: 0.91</li><li>Accessibility: 0.98</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/locale/:</strong></p><ul><li>Performance: 0.92</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr></details>

File: src/components/modals.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+import {
+  Button,
+  ButtonVariant,
+  CloseModalCrossButton,
+} from "./buttons/buttons";
+import { ModalOverlay } from "./modal-overlay";
+
+enum ModalVariant {
+  MARK_AS_PRIMARY = "MARK_AS_PRIMARY",
+  DELETION = "DELETION",
```
Comment (Meyclem) [inline src/components/modals.tsx:12]: I find the naming a bit to specific, maybe something like: `ACTION` and `DESTRUCTIVE_ACTION` (or like so)?

File: tests/unit/modals.test.tsx [hunk 3]
```diff
@@ -135,60 +139,51 @@ describe("ConfirmationBox", () => {
...
-  //   await screen
-  //     .findByText(
-  //       new RegExp(
-  //         `You are about to delete ${mockIdentities.nonPrimaryEmailIdentity.value}`,
-  //         "i",
-  //       ),
-  //     )
-  //     .then((waitedElement) => {
-  //       expect(waitedElement).not.toBeVisible();
+  //   it("should close confirmation box when clicking outside of it", async () => {
```
Comment (Meyclem) [inline tests/unit/modals.test.tsx:154]: Do you want to leave this commented code?

**Pull request merged**